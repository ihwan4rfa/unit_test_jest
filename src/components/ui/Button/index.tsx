import React from "react";
import { ButtonProps } from "./types.type";

const Button = (props: ButtonProps) => {
  const { children, onClick, variant = "primary" } = props;
  return (
    <button className={variant} onClick={() => onClick()}>
      {children}
    </button>
  );
};

export default Button;
