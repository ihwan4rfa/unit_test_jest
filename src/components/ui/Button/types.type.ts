export type ButtonProps = {
  children: React.ReactNode;
  onClick: () => void;
  variant?: string;
};
