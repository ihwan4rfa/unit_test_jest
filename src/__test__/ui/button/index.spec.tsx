import React from "react";
import { fireEvent, render } from "@testing-library/react";
// import "@testing-library/jest-dom/extend-expect";
import Button from "@/components/ui/Button";

describe("button component", () => {
  it("should render button component", () => {
    const { container } = render(
      <Button onClick={() => console.log("clicked")}>Save</Button>
    );
    expect(container).toMatchSnapshot();
  });
  it("should click onclick event", () => {
    const { getByText } = render(
      <Button onClick={() => console.log("clicked")}>Save</Button>
    );
    const buttonSubmit = getByText("Submit");
    fireEvent.click(buttonSubmit);
    expect(buttonSubmit).toBeInTheDocument();
  });
});
