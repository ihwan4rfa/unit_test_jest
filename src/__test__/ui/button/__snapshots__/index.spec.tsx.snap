// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should render button component 1`] = `
<div>
  <button
    class="primary"
  >
    Save 
  </button>
</div>
`;
